tests/evaluate/test_default_evaluator.py

```diff
@@ -3192,6 +3193,149 @@ def test_evaluate_custom_metrics_string_values():
+    assert "eval_results_table.json" in artifacts
+    pd.DataFrame(**results.artifacts["eval_results_table"].content)
+    # TODO: fix the logged data to name the model output column "retrieved_context"
+    # Right now, it's hard-coded "outputs", which is not ideal
+    # validate_retriever_logged_data(logged_data)
+    assert set(results.metrics.keys()) == {
+        "precision_at_k/v1/p90",
+        "precision_at_k/v1/mean",
+        "precision_at_k/v1/variance",
+    }
+    assert results.metrics["precision_at_k/v1/p90"] == 2 / 3
+    assert results.metrics["precision_at_k/v1/mean"] == 2 / 3
+    assert results.metrics["precision_at_k/v1/variance"] == 0
+
+    # test with a big k to ensure we use min(k, len(retrieved_chunks))
+    with mlflow.start_run() as run:
+        mlflow.evaluate(
+            model=fn,
+            data=X,
+            targets="ground_truth",
+            model_type="retriever",
+            evaluators="default",
+            evaluator_config={
+                "k": 6,
+            },
+        )
+    run = mlflow.get_run(run.info.run_id)
+    assert run.data.metrics == {
+        "precision_at_k/v1/mean": 2 / 3,
+        "precision_at_k/v1/variance": 0,
+        "precision_at_k/v1/p90": 2 / 3,
+    }
+
+    # test with multiple chunks from same doc
+    def fn2(X):
+        return pd.DataFrame({"retrieved_context": [("doc1", "doc1", "doc3")] * len(X)})
+
+    X = pd.DataFrame({"question": ["q1?"] * 3, "ground_truth": [("doc1", "doc3")] * 3})
+
+    with mlflow.start_run() as run:
+        mlflow.evaluate(
+            model=fn2,
+            data=X,
+            targets="ground_truth",
+            model_type="retriever",
+            evaluator_config={
+                "default": {
+                    "k": 3,
+                }
+            },
+        )
+    run = mlflow.get_run(run.info.run_id)
+    assert run.data.metrics == {
+        "precision_at_k/v1/mean": 1,
+        "precision_at_k/v1/variance": 0,
+        "precision_at_k/v1/p90": 1,
+    }
+
+    # test with empty returned chunks
+    def fn3(X):
+        return pd.DataFrame({"output": [()] * len(X)})
+
+    with mlflow.start_run() as run:
+        mlflow.evaluate(
+            model=fn3,
+            data=X,
+            targets="ground_truth",
+            model_type="retriever",
+            evaluator_config={
+                "default": {
+                    "k": 3,
+                }
+            },
+        )
+    run = mlflow.get_run(run.info.run_id)
+    assert run.data.metrics == {
+        "precision_at_k/v1/mean": 1,
+        "precision_at_k/v1/variance": 0,
+        "precision_at_k/v1/p90": 1,
+    }
+
+
+def test_evaluate_retriever_error_handling():
+    X = pd.DataFrame({"question": ["q1?"] * 3, "ground_truth": [("doc1", "doc2")] * 3})
+
+    def fn(X):
+        return pd.DataFrame({"output": [("doc1", "doc3", "doc2")] * len(X)})
+
+    # TODO: improve error message, avoid saying "columns ['k']"
+    with pytest.raises(
```

Comment:
Do we want to fail/raise an exception if 'precision_at_k' is not configured correctly? For now (other builtin metrics), we don't fail if they are missing columns (e.x. targets) but log that the metric failed to compute and continue - we only fail for extra_metrics. This could be different considering there is only one metric for retriever model type right now, just wanted to mention that this behavior differs from other builtin metrics.

Reply:
Good point! This will not be a problem after we switch to https://github.com/mlflow/mlflow/pull/10079/files#r1369217765.

Reply:
Updated to no longer expecting an error.